cmd/identityoracled/handlers/identity.go

```diff
@@ -90,11 +99,18 @@ func (v *Verify) XPubSignature(ctx context.Context, log logger.Logger, w http.Re
 		return translate(errors.Wrap(err, "unmarshal request"))
 	}
 
+	for _, xpub := range requestData.XPubs {
+		if xpub.IsPrivate() {
+			web.Respond(ctx, log, w, "private keys not allowed", http.StatusUnprocessableEntity)
```

Comment:
`return web.Respond(...`

Reply:
This web.Respond doesn't return an error. It creates a response no matter what.